crates/fluvio-cli/src/connector/update.rs

```diff
@@ -0,0 +1,87 @@
+//!
+//! # Update a Managed Connector
+//!
+//! CLI tree to generate Update a Managed Connector
+//!
+
+use fluvio::metadata::topic::ReplicaSpec;
+use structopt::StructOpt;
+use tracing::debug;
+
+use fluvio::{Fluvio, FluvioError};
+use fluvio::metadata::{
+    topic::{TopicSpec, TopicReplicaParam},
+    connector::ManagedConnectorSpec,
+};
+
+use fluvio_sc_schema::ApiError;
+use fluvio_sc_schema::errors::ErrorCode;
+
+use crate::CliError;
+use super::ConnectorConfig;
+
+// -----------------------------------
+// CLI Options
+// -----------------------------------
+
+#[derive(Debug, StructOpt, Default)]
+pub struct UpdateManagedConnectorOpt {
+    /// The name for the update Managed Connector
+    #[structopt(short = "c", long = "config", value_name = "config", required(true))]
+    pub config: String,
+}
+
+impl UpdateManagedConnectorOpt {
+    pub async fn process(self, fluvio: &Fluvio) -> Result<(), CliError> {
+        let config = ConnectorConfig::from_file(&self.config)?;
+        let spec: ManagedConnectorSpec = config.clone().into();
+        let name = spec.name.clone();
+
+        debug!("updating managed_connector: {}, spec: {:#?}", name, spec);
+
+        let admin = fluvio.admin().await;
+
+        // admin.find ?
+        let lists = admin.list::<ManagedConnectorSpec, _>(vec![]).await?;
+        let existing_config = match lists.into_iter().find(|c| c.name == config.name) {
+            None => {
+                return Err(CliError::InvalidConnector(format!(
+                    "{} - does not exist.",
+                    config.name
+                )))
+            }
+            Some(c) => c,
+        };
+
+        debug!("Found connector: {:?}", existing_config);
+
+        // create_topic will succeed even if the topic already exists. PR: 1823
+        // A New Topic might have been defined - We will not delete the old one.
+        if config.create_topic {
+            // Check if the topic already exists by trying to create it
+            let replica_spec = ReplicaSpec::Computed(TopicReplicaParam::new(1, 1, false));
+            debug!("UpdateManagedConnectorOpt topic spec: {:?}", replica_spec);
+            match admin
+                .create::<TopicSpec>(config.topic, false, replica_spec.into())
+                .await
+            {
+                Err(FluvioError::AdminApi(ApiError::Code(ErrorCode::TopicAlreadyExists, _))) => {
+                    debug!("UpdateManagedConnectorOpt - Re-using Topic");
+                    Ok(())
+                }
+                Ok(_) => {
+                    debug!("UpdateManagedConnectorOpt - Created a new topic.");
+                    Ok(())
+                }
+                Err(e) => Err(e),
+            }?;
+        }
+
+        admin
+            .delete::<ManagedConnectorSpec, _>(&config.name)
+            .await?;
+        admin.create(config.name, false, spec).await?;
```

Comment:
Hmm. So, this takes a new config, delete's the one with the same name and then recreates it. While technically this does update the `ManagedConnectorSpec`, if we want resume-from-last-offset using persistent storage, this won't work as the pod will be deleted.

I think we might need an [`UpdateRequest` in the public API](https://github.com/infinyon/fluvio/blob/0c5560a3fa1f6e024896dd0801878d16fa5150d1/crates/fluvio-sc/src/services/public_api/public_server.rs#L80-L119). @sehz What do you think?

Reply:
Or we can rename, `CreateRequest` as `CreateOrUpdateRequest`.   This can be mapped to Kubernetes `apply`.

Reply:
I'm okay with that. The only thing there is that all our `CreateRequests` would then support updates where as now if you try and create the same topic, it errors saying "topic already exists".

Reply:
We decided to add update request into API